### PR TITLE
remove spurious load nodes for address-of-field events

### PR DESCRIPTION
**Pull request by aneksteind**
Addresses [this bug](https://github.com/immunant/c2rust/pull/564#discussion_r934787433)

### CODE REVIEW

File: pdg/src/snapshots/c2rust_pdg__tests__analysis_test_pdg_snapshot.snap [hunk 1]
```diff
@@ -108,28 +107,26 @@ g {
 	n[12]: field.0    n[3]  => _12 @ bb4[18]: fn simple; _12 = ((*_1).0: i32)                        
 	n[13]: addr.load  n[12] => _   @ bb4[18]: fn simple; _12 = ((*_1).0: i32)                        
 	n[14]: field.1    n[3]  => _13 @ bb4[21]: fn simple; _13 = &raw const ((*_1).1: u64)             
-	n[15]: addr.load  n[14] => _   @ bb4[21]: fn simple; _13 = &raw const ((*_1).1: u64)             
-	n[16]: addr.load  n[3]  => _   @ bb4[24]: fn simple; _14 = &raw const (*_1)
```
Comment (spernsteiner) [inline pdg/src/snapshots/c2rust_pdg__tests__analysis_test_pdg_snapshot.snap:112]: `_14 = &(*_1)` should produce a `Copy` node, shouldn't it?

Comment (aneksteind) [reply pdg/src/snapshots/c2rust_pdg__tests__analysis_test_pdg_snapshot.snap:112]: Yes, that was fixed in #533 but this wasn't up to date with master. Now fixed.

**Comment by kkysen** [comment]
@aneksteind, sorry, I was trying to merge, but thought I was on a different PR/branch, and accidentally included an extra change in the merge commit (https://github.com/immunant/c2rust/pull/565/commits/a3b2a29b36d223f11888a1067753317fda5e7f0d).  I undid that in https://github.com/immunant/c2rust/pull/565/commits/b6118b79e998ee033dd17f1f244341cc245cf06d, but I'm unsure how to squash that back into the merge commit.  The merge is otherwise fine, as I just updated the snapshot.

**Comment by spernsteiner** [comment]
@kkysen,

>I'm unsure how to squash that back into the merge commit

`git rebase -ir HEAD~2` for an interactive rebase that preserves the branch structure of the merge.  Then just change the last commit from `pick` to `fixup`.

I think `git reset` to the merge commit and `git commit --amend` would also work.

**Comment by kkysen** [comment]
Thanks, @spernsteiner, I think that worked!

File: pdg/src/snapshots/c2rust_pdg__tests__analysis_test_pdg_snapshot.snap (hunk 1 above)
Comment (spernsteiner) [reply pdg/src/snapshots/c2rust_pdg__tests__analysis_test_pdg_snapshot.snap]: The `LoadAddr` is gone now, but there's still no `Copy` at `bb4[24]` in the snapshot

File: pdg/src/snapshots/c2rust_pdg__tests__analysis_test_pdg_snapshot.snap [hunk 2]
```diff
@@ -87,51 +87,50 @@ g {
...
+	n[9]:  addr.store  n[8]  => _      @ bb4[14]: fn simple; ((*_5).0: i32) = move _11;               
+	n[10]: field.1     n[3]  => _      @ bb4[16]: fn simple; ((*_1).1: u64) = const 9_u64;            
+	n[11]: addr.store  n[10] => _      @ bb4[16]: fn simple; ((*_1).1: u64) = const 9_u64;            
+	n[12]: field.0     n[3]  => _12    @ bb4[18]: fn simple; _12 = ((*_1).0: i32);                    
+	n[13]: addr.load   n[12] => _      @ bb4[18]: fn simple; _12 = ((*_1).0: i32);                    
+	n[14]: field.1     n[3]  => _13    @ bb4[21]: fn simple; _13 = &raw const ((*_1).1: u64);         
+	n[15]: copy        n[3]  => _14    @ bb4[25]: fn simple; _14 = &raw const (*_1);                  
+	n[16]: field.2     n[3]  => _      @ bb4[25]: fn simple; ((*_1).2: *const pointers::S) = move _14;
+	n[17]: addr.store  n[16] => _      @ bb4[25]: fn simple; ((*_1).2: *const pointers::S) = move _14;
+	n[18]: value.store n[15] => _1.*.2 @ bb4[26]: fn simple; ((*_1).2: *const pointers::S) = move _14;
```
Comment (spernsteiner) [inline pdg/src/snapshots/c2rust_pdg__tests__analysis_test_pdg_snapshot.snap:115]: Something must have gone wrong here - it's showing two different statements at the same position:
```
bb4[25]: fn simple; _14 = &raw const (*_1);                  
bb4[25]: fn simple; ((*_1).2: *const pointers::S) = move _14;
```
Followed by the same statement at two different positions:
```
bb4[25]: fn simple; ((*_1).2: *const pointers::S) = move _14;
bb4[26]: fn simple; ((*_1).2: *const pointers::S) = move _14;
```

Comment (aneksteind) [reply pdg/src/snapshots/c2rust_pdg__tests__analysis_test_pdg_snapshot.snap:115]: Made an issue here https://github.com/immunant/c2rust/issues/581

**Review by spernsteiner — APPROVED** [review]

**Pull request merged**